### ISSUE TITLE
Code for Mac opens too many files, rendering other apps unusable

### ISSUE THREAD

**Issue by ftrofin** [Microsoft/vscode#43804]
- VSCode Version: 1.20.1
- OS Version: 10.12.6
- C/C++ extension version  0.15.0

Steps to Reproduce:

1. Launch VS Code on Mac.
2. Open a project/folder that has a lot of C/C++ files (517386 in my case)
3. try to do other work on the system (like compile a project in Xcode).
4. Various things will fail as the max number of open files is exhausted (For example Terminal will bail out with an error, Xcode build will fail, etc).

NOTE: the above happens even if this is not the first time opening that folder (so it was already indexed by VSCode).

ulimit for open files is set to 4096 but the problem still persists:
$ ulimit -a
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
file size               (blocks, -f) unlimited
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) 4096
pipe size            (512 bytes, -p) 1
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 709
virtual memory          (kbytes, -v) unlimited

I really really like VSCode and I would like to use it instead of Xcode but unfortunately this issue is a blocker.

Does this issue occur when all extensions are disabled?: 
- I don't know but I really need the C++ extension enabled, otherwise VSCode is useless for me.

Please let me know if you need any more info.

**Comment by bpasero** [comment]
I wonder if this issue is being caused by an installed extension. Can you try to run VS Code without extensions? From the command line (NOT the integrated terminal in Code), execute: `code --disable-extensions` and try your steps again to see if it reproduces. If you see it is an issue with the extension, please file it against the extension repository itself.

**Comment by ftrofin** [comment]
It might be, i couldn't verify. I disabled all the extensions and did not happened. I enabled them all again and did not happen for a while. I disabled the C++ extension and still did not happened. I re-enabled it and waiting for it since...
Since  I have quite a number of apps open on my computer, I decided to try to increase the file descriptor limit by adding a plist to the /Library/LaunchDaemons (see instructions here for various OS versions including Sierra: https://superuser.com/questions/302754/increase-the-maximum-number-of-open-file-descriptors-in-snow-leopard)
I'm going to close this issue as I don't have proof that Code was at fault here.
